xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/script/RefactoringScriptService.java

```diff
@@ -102,181 +103,21 @@
-        return request;
-    }
-
-    /**
-     * Creates a request to copy the specified entity with a different name.
-     *
-     * @param reference the entity to copy
-     * @param copyName the name of the entity copy
-     * @return the copy-as request
-     */
-    public MoveRequest createCopyAsRequest(EntityReference reference, String copyName)
-    {
-        EntityReference copyReference = new EntityReference(copyName, reference.getType(), reference.getParent());
-        return createCopyAsRequest(reference, copyReference);
-    }
-
-    /**
-     * Creates a request to delete the specified entities.
-     *
-     * @param entityReferences the entities to delete
-     * @return the delete request
-     */
-    public EntityRequest createDeleteRequest(Collection<EntityReference> entityReferences)
-    {
-        EntityRequest request = new EntityRequest();
-        initEntityRequest(request, RefactoringJobs.DELETE, entityReferences);
-        return request;
-    }
-
-    /**
-     * Creates a request to create the specified entities.
-     *
-     * @param entityReferences the entities to create
-     * @return the create request
-     * @since 7.4M2
-     */
-    public CreateRequest createCreateRequest(Collection<EntityReference> entityReferences)
-    {
-        CreateRequest request = new CreateRequest();
-        initEntityRequest(request, RefactoringJobs.CREATE, entityReferences);
-        // Set deep create by default, to copy (if possible) any existing hierarchy of a specified template document.
-        // TODO: expose this in the create UI to advanced users to allow them to opt-out?
-        request.setDeep(true);
-        return request;
-    }
-
-    private void initEntityRequest(EntityRequest request, String type, Collection<EntityReference> entityReferences)
-    {
-        request.setId(generateJobId(type));
-        request.setJobType(type);
-        request.setEntityReferences(entityReferences);
-        setRightsProperties(request);
-    }
-
-    private MoveRequest createMoveRequest(String type, Collection<EntityReference> sources, EntityReference destination)
-    {
-        MoveRequest request = new MoveRequest();
-        initEntityRequest(request, type, sources);
-        request.setDestination(destination);
-        request.setUpdateLinks(true);
-        request.setAutoRedirect(true);
-        request.setUpdateParentField(true);
-        return request;
-    }
+    private RequestFactory requestFactory;
 
-    /**
-     * @param type the type of refactoring
-     * @return an id for a job to perform the specified type of refactoring
-     */
-    private List<String> generateJobId(String type)
+    @Override
+    public void initialize() throws InitializationException
     {
-        String suffix = new Date().getTime() + "-" + ThreadLocalRandom.current().nextInt(100, 1000);
-        return getJobId(type, suffix);
+        this.requestFactory = new RequestFactory(documentAccessBridge, modelContext);
```

Comment:
Would be cleaner to make RequestFactory and inject it.

Reply:
Actually we discussed it with @vmassol and it seemed more interesting to keep it like that: we don't really need it as a component since we only need inside RefactoringScriptService. And it only needs two external components for now.

Reply:
> we only need inside RefactoringScriptService

It does not really matter IMO.

Another argument is that RefactoringScriptService does not use those components so it will reduce the ClassFanOutComplexity.

Reply:
I'm still against since it would mean adding a new interface that we don't really need here. So I'd stay with this for now.

Reply:
Oh. Then I indeed don't really see a reason why not using a component. I'll change that.

Reply:
Of course it also need to be added in components.txt ;)